source/extensions/quic_listeners/quiche/quic_filter_manager_connection_impl.cc

```diff
@@ -0,0 +1,118 @@
+#include "extensions/quic_listeners/quiche/quic_filter_manager_connection_impl.h"
+
+#include <memory>
+
+namespace Envoy {
+namespace Quic {
+
+QuicFilterManagerConnectionImpl::QuicFilterManagerConnectionImpl(
+    std::unique_ptr<EnvoyQuicConnection> connection, Event::Dispatcher& dispatcher)
+    : quic_connection_(std::move(connection)), filter_manager_(*this), dispatcher_(dispatcher),
+      // QUIC connection id can be 18 bytes. It's easier to use hash value instead
+      // of trying to map it into a 64-bit space.
+      stream_info_(dispatcher.timeSource()), id_(quic_connection_->connection_id().Hash()) {
+  // TODO(danzh): Use QUIC specific enum value.
+  stream_info_.protocol(Http::Protocol::Http2);
+}
+
+void QuicFilterManagerConnectionImpl::addWriteFilter(Network::WriteFilterSharedPtr filter) {
+  filter_manager_.addWriteFilter(filter);
+}
+
+void QuicFilterManagerConnectionImpl::addFilter(Network::FilterSharedPtr filter) {
+  filter_manager_.addFilter(filter);
+}
+
+void QuicFilterManagerConnectionImpl::addReadFilter(Network::ReadFilterSharedPtr filter) {
+  filter_manager_.addReadFilter(filter);
+}
+
+bool QuicFilterManagerConnectionImpl::initializeReadFilters() {
+  return filter_manager_.initializeReadFilters();
+}
+
+void QuicFilterManagerConnectionImpl::addConnectionCallbacks(Network::ConnectionCallbacks& cb) {
+  network_connection_callbacks_.push_back(&cb);
+}
+
+void QuicFilterManagerConnectionImpl::enableHalfClose(bool enabled) {
+  ASSERT(!enabled, "Quic connection doesn't support half close.");
+}
+
+void QuicFilterManagerConnectionImpl::setBufferLimits(uint32_t /*limit*/) {
+  // TODO(danzh): add interface to quic for connection level buffer throttling.
+  // Currently read buffer is capped by connection level flow control. And
+  // write buffer is not capped.
+  ENVOY_CONN_LOG(error, "Quic manages its own buffer currently.", *this);
+}
+
+void QuicFilterManagerConnectionImpl::close(Network::ConnectionCloseType type) {
+  if (type != Network::ConnectionCloseType::NoFlush) {
+    // TODO(danzh): Implement FlushWrite and FlushWriteAndDelay mode.
+    ENVOY_CONN_LOG(error, "Flush write is not implemented for QUIC.", *this);
```

Comment:
Same here? Basically can we remove the error logs and just crash or have a TODO?

Reply:
We don't want to crash, as it's called by HCM with other CloseType. Removed the error log. The empty if block looks weird though...